server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java

```diff
@@ -2244,6 +2247,26 @@ protected boolean supportsEmptyInputArray() {
         return false;
     }
 
+    @Override
+    protected BlockReaderSupport getSupportedReaders(MapperService mapper, String loaderFieldName) {
+        return new BlockReaderSupport(true, true, mapper, loaderFieldName);
+    }
+
+    @Override
+    protected Function<Object, Object> loadBlockExpected(BlockReaderSupport blockReaderSupport, boolean columnReader) {
+        DenseVectorFieldType ft = (DenseVectorFieldType) blockReaderSupport.mapper().fieldType(blockReaderSupport.loaderFieldName());
+        if (ft.getElementType() != ElementType.FLOAT) {
+            return null;
+        }
+
+        return Function.identity();
```

Comment:
nit (optional) `return ft.getElementType() == ElementType.FLOAT ? Function.identity() : null;`

Reply:
I made the if stand out because we will get rid of that in the future 😉